test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,335 @@
+        size_t rpos = random_read_pos(gen);
+        std::vector<alphabet_t> read_tmp{std::ranges::begin(ref) + rpos,
+                                         std::ranges::begin(ref) + rpos + read_length + simulated_errors};
+
+        // generate simulated_errors many unique random mutation positions
+        std::set<size_t> mutation_positions;
+        if (read_length > simulated_errors){
+            while (mutation_positions.size() < simulated_errors)
+                mutation_positions.insert(random_mutation_pos(gen));
+        }
+        else
+        {
+            for(size_t i = 0; i < simulated_errors; ++i)
+                mutation_positions.insert(i);
+        }
+
+        for (std::set<size_t>::iterator pos_it = mutation_positions.begin(); pos_it != mutation_positions.end(); ++pos_it)
+        {
+            size_t ppos = *pos_it;
+            double prob = mutation_type_prob(gen);
+            // Substitution
+            if (prob_insertion + prob_deletion < prob)
+                mutate_substitution(read_tmp, ppos, dis_alpha_short(gen));
+            // Insertion
+            else if (prob_insertion < prob)
+                mutate_insertion(read_tmp, ppos, dis_alpha(gen));
+            // Deletion
+            else
+                mutate_deletion(read_tmp, ppos);
+        }
+
+        read_tmp.erase(std::ranges::begin(read_tmp) + read_length, std::ranges::end(read_tmp));
+        reads.push_back(read_tmp);
+    }
+
+    return reads;
+}
+
+template <typename alphabet_t>
+std::vector<alphabet_t> generate_repeating_sequence(size_t const template_length = 5000,
+                                                    size_t const repeats = 20,
+                                                    double const template_fraction = 1,
+                                                    size_t const seed = 0)
+{
+    std::vector<alphabet_t> seq_template = generate_sequence<alphabet_t>(template_length, 0, seed);
+
+    // copy substrings of length len from seq_template mutate and concatenate them
+    size_t len = std::round(template_length * template_fraction);
+    uint8_t simulated_errors = 5;
+    len = (len + simulated_errors  > template_length) ? template_length - simulated_errors : len;
+
+    return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | view::persist | std::view::join  | std::ranges::to<std::vector>;
+}
+
+//============================================================================
+//  undirectional; trivial_search, collection, dna4, all-mapping
+//============================================================================
+
+void unidirectional_search_all_collection(benchmark::State & state, options && o)
+{
+    size_t set_size = 10;
+    std::vector<std::vector<seqan3::dna4>> collection;
+    std::vector<std::vector<seqan3::dna4>> reads;
+    for (size_t i = 0; i < set_size; ++i)
+    {
+        collection.push_back(generate_sequence<seqan3::dna4>(o.sequence_length, 0, i));
```

Comment:
During the generation of a collection of sequences, the same seed was used for each sequence (resulting in the same sequence). This was also the case for the read generation.